lib/src/migrators/division.dart

```diff
@@ -0,0 +1,208 @@
+// Copyright 2018 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:args/args.dart';
+
+// The sass package's API is not necessarily stable. It is being imported with
+// the Sass team's explicit knowledge and approval. See
+// https://github.com/sass/dart-sass/issues/236.
+import 'package:sass/src/ast/sass.dart';
+
+import 'package:sass_migrator/src/migration_visitor.dart';
+import 'package:sass_migrator/src/migrator.dart';
+import 'package:sass_migrator/src/patch.dart';
+import 'package:sass_migrator/src/utils.dart';
+
+/// Migrates stylesheets that use the `/` operator for division to use the
+/// `divide` function instead.
+class DivisionMigrator extends Migrator {
+  final name = "division";
+  final description =
+      "Migrates from the / division operator to the divide function";
+
+  @override
+  final argParser = ArgParser()
+    ..addFlag('aggressive',
+        abbr: 'a',
+        help: r"If set, expressions like '$x/3 + 1', '$x/3 - 1', and 'fn()/3' "
+            "will be migrated.");
+
+  bool get isAggressive => argResults['aggressive'] as bool;
+
+  Map<Uri, String> migrateFile(Uri entrypoint) =>
+      _DivisionMigrationVisitor(this.isAggressive, migrateDependencies)
+          .run(entrypoint);
+}
+
+class _DivisionMigrationVisitor extends MigrationVisitor {
+  final bool isAggressive;
+
+  _DivisionMigrationVisitor(this.isAggressive, bool migrateDependencies)
+      : super(migrateDependencies: migrateDependencies);
+
+  /// True when division is allowed by the context the current node is in.
+  bool _isDivisionAllowed = false;
+
+  /// True when the current node is expected to evaluate to a number.
+  bool _expectsNumericResult = false;
+
+  /// If this is a division operation, migrates it.
+  ///
+  /// If this is any other operator, allows division within its left and right
+  /// operands.
+  @override
+  void visitBinaryOperationExpression(BinaryOperationExpression node) {
+    if (node.operator == BinaryOperator.dividedBy) {
+      if (shouldMigrate(node)) {
+        addPatch(patchBefore(node, "divide("));
+        patchSlashToComma(node);
+        addPatch(patchAfter(node, ")"));
+      }
+      super.visitBinaryOperationExpression(node);
+    } else {
+      withContext(
+          true,
+          _expectsNumericResult || operatesOnNumbers(node.operator),
+          () => super.visitBinaryOperationExpression(node));
+    }
+  }
```

Comment:
I think `/` operations that aren't migrated to division should be migrated to the `slash-list()` function ([described here](https://github.com/sass/sass/issues/2565#issue-362710498)), at least in aggressive mode. This will ensure that users' runtime behavior won't change when `/` is changed to return a list rather than a string.

Reply:
Should we do this for the ambiguous slashes that we output warnings for or just the slashes that are unambiguously not division?